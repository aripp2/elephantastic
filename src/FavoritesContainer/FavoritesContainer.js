import React from 'react';
import './FavoritesContainer.scss';

const FavoritesContainer = () => {
  return (
    <section>Favorites</section>
  )
}

export default FavoritesContainer;