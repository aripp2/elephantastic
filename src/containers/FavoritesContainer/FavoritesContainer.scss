@import './src/variables.scss';


.favs-container {
  padding: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.img-container {
  width: 30%;
  margin: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.fav-img {
  height: auto; 
  width: auto; 
  max-width: 400px; 
  max-height: 400px;
  border-radius: 5px;
  // position: relative;
}

.remove-btn {
  height: 30px;
  // width: auto;
  font-family: $body-font;

  font-size: 0.8em;
  border-radius: 0 0 5px 5px;
  &:hover {
    color: white;
    font-weight: 700;
    background-color: black;
  }
}