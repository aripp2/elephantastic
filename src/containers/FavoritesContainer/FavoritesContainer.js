import React from 'react';
import { connect } from 'react-redux';
import './FavoritesContainer.scss';

export const FavoritesContainer = ({ favorites, updateFavs }) => {

  const favs = favorites.map(fav => {
    return (
      <div className='img-container' key={fav.image_id}>
        <img 
          className='fav-img'
          id={fav.id} 
          src={fav.image.url} 
          alt='dog' 
        />
        <button
          className='remove-btn'
          onClick={() => updateFavs(fav.image_id, true, fav.id)}
        >Delete Favorite</button>
      </div>
     ) 
  })

  return (
    <section className='favs-container'>
      {favs}
    </section>
  )
}

export const mapStateToProps = ({ favorites }) => ({
  favorites
})

export default connect(mapStateToProps)(FavoritesContainer);