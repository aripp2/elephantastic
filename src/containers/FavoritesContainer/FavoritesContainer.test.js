import React from 'react';
import { shallow } from 'enzyme';
import { FavoritesContainer, mapStateToProps } from './FavoritesContainer';

const mockFavs = [
    {
      created_at: "2019-11-01T17:26:47.000Z",
      id: 1095,
      image: {id: "ry8KWgqEQ", url: "https://cdn2.thedogapi.com/images/ry8KWgqEQ.jpg"},
      image_id: "ry8KWgqEQ",
      sub_id: null,
      user_id: "6xrk1n"
    },
    {
      created_at: "2019-11-02T18:32:57.000Z",
      id: 1098,
      image: {id: "edNIMgpSw", url: "https://cdn2.thedogapi.com/images/edNIMgpSw.jpg"},
      image_id: "edNIMgpSw",
      sub_id: null,
      user_id: "6xrk1n"
    }
  ];

  describe('FavoritesContainer', () => {

    let wrapper;
    const mockUpdateFavs = jest.fn();

    beforeEach(() => {
      wrapper = shallow(<FavoritesContainer
          favorites={mockFavs} 
          updateFavs={mockUpdateFavs}
        />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call updateFavs when button is clicked', () => {
      wrapper.find('button').at(0).simulate('click');
      expect(mockUpdateFavs).toHaveBeenCalledWith('ry8KWgqEQ', true, 1095);
    });
  });

  describe('mapStateToProps', () => {

  it('should return an object of favs array', () => {
    const mockState = {
      favorites: mockFavs,
      randomPup: {},
      errorMsg: '',
      isLoading: false,
      breeds: []
    };
    const expected = {
      favorites: mockFavs
    }
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});