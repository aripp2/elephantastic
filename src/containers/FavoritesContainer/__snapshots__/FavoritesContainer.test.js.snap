// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer should match the snapshot 1`] = `
<section>
  <div
    key="ry8KWgqEQ"
  >
    <img
      alt="dog"
      id={1095}
      src="https://cdn2.thedogapi.com/images/ry8KWgqEQ.jpg"
    />
    <button
      onClick={[Function]}
    >
      Delete Favorite
    </button>
  </div>
  <div
    key="edNIMgpSw"
  >
    <img
      alt="dog"
      id={1098}
      src="https://cdn2.thedogapi.com/images/edNIMgpSw.jpg"
    />
    <button
      onClick={[Function]}
    >
      Delete Favorite
    </button>
  </div>
</section>
`;
